### PR TITLE
Display Flush fix plus extra SPI displays

### PR DESCRIPTION
**Pull request by AdrianSoundy**
## Description

Changes to the BitBlt method to enable it to work with a window in the bitmap.
Tidied up some code.

Also added initial drivers for SPI displays SSD1331(colour) and SSD1306(mono)
These displays only require a small frame buffer so they can work on an ESP32 without spiram. The SSD1306 also comes in I2C version but held back I2C driver as that requires other changes to share bus and have a generic I2C interface for alll platforms.

## Motivation and Context
- Fixes nanoFramework/Home#655

## How Has This Been Tested?

Tested using small test program that displayed partial parts of complete screen and also tested with Tetris sample for larger screens( to be updated)

Following combinations tested
- STM32F769I_DISCOVERY with in built in display
- ESP32 Wrover V4 with ILI9341 display
- ESP32 Wrover V3 with ST7789V display
- ESP32 devkit with SSD1331 94x64 colour
- ESP32 devkit with SPI SSD1306 128x64 mono

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/nanoFramework.Graphics/Graphics/Displays/Spi_To_Display.cpp [hunk 1]
```diff
@@ -92,57 +104,109 @@ void DisplayInterface::SendCommand(CLR_UINT8 arg_count, ...)
...
+    }
+    else
+    {
+        // Send all Command bytes with D/C signal low
+        SendBytes(parameters, arg_count);
+        CPU_GPIO_SetPinState(lcdDC, GpioPinValue_High); // Data mode
+    }
 }
 
+// void DisplayInterface::SendCommand2(CLR_UINT8 arg_count, ...)
```
Comment (josesimoes) [inline src/nanoFramework.Graphics/Graphics/Displays/Spi_To_Display.cpp:136]: Is this commented code still needed here?

**Review by josesimoes — COMMENTED** [review]
All the rest LGTM!

File: src/nanoFramework.Graphics/Graphics/Displays/Spi_To_Display.cpp (hunk 1 above)
Comment (AdrianSoundy) [reply src/nanoFramework.Graphics/Graphics/Displays/Spi_To_Display.cpp:136]: opps

Comment (AdrianSoundy) [reply src/nanoFramework.Graphics/Graphics/Displays/Spi_To_Display.cpp]: Fixed now

**Review by josesimoes — APPROVED** [review]
LGTM!

**Pull request merged**